compiler/qsc_frontend/src/typeck/check.rs

```diff
@@ -208,21 +208,28 @@ impl Visitor<'_> for ItemCollector<'_> {
                 self.checker.globals.insert(item, scheme);
             }
             ast::ItemKind::Ty(name, def) => {
+                let span = item.span;
                 let Some(&Res::Item(item)) = self.names.get(name.id) else {
                     panic!("type should have item ID");
                 };
 
-                let (base, base_errors) = convert::ast_ty_def_base(self.names, def);
                 let (cons, cons_errors) = convert::ast_ty_def_cons(self.names, item, def);
+                let (udt_def, def_errors) = convert::ast_ty_def(self.names, def);
                 self.checker.errors.extend(
-                    base_errors
+                    cons_errors
                         .into_iter()
-                        .chain(cons_errors)
+                        .chain(def_errors)
                         .map(|MissingTyError(span)| Error(ErrorKind::MissingItemTy(span))),
                 );
```

Comment:
nitpick: the order of the error reporting changed here, since now `cons_errors` will get reported first. They are all missing type errors, so it probably doesn't make a difference, just wanted to know if that was intentional?

Reply:
This should surface different errors. Before, the `ast_ty_def_base` call and the `ast_ty_def_cons` call would yield the same error twice, as `ast_ty_def_base` is already called by `ast_ty_def_cons`. The `def_errors` are different errors that are created when we try to convert the ast types in the UDT def to hir types.